crates/router/src/connector/dummyconnector.rs

```diff
@@ -136,10 +141,22 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
 
     fn get_url(
         &self,
-        _req: &types::PaymentsAuthorizeRouterData,
-        _connectors: &settings::Connectors,
+        req: &types::PaymentsAuthorizeRouterData,
+        connectors: &settings::Connectors,
     ) -> CustomResult<String, errors::ConnectorError> {
-        Err(errors::ConnectorError::NotImplemented("get_url method".to_string()).into())
+        let payment_method_data = req.request.get_payment_method_data();
```

Comment:
```suggestion
        let payment_method_data = req.request.payment_method_data;
```

When you want to throw some error if the data is not present then use getter methods otherwise use the direct data